node/modules/chain.go

```diff
@@ -34,8 +34,8 @@ import (
 func ChainBitswap(lc fx.Lifecycle, mctx helpers.MetricsCtx, host host.Host, rt routing.Routing, bs dtypes.ExposedBlockstore) dtypes.ChainBitswap {
 	// prefix protocol for chain bitswap
 	// (so bitswap uses /chain/ipfs/bitswap/1.0.0 internally for chain sync stuff)
-	bitswapNetwork := network.NewFromIpfsHost(host, rt, network.Prefix("/chain"))
-	bitswapOptions := []bitswap.Option{bitswap.ProvideEnabled(false)}
+	bitswapNetwork := bsnet.NewFromIpfsHost(host, bsnet.Prefix("/chain"))
+	bitswapOptions := []bitswap.Option{bitswap.WithServerEnabled(false)}
```

Comment:
The AI seems to have added this option that disables the bitswap server which will basically kill the network and should explain the failing tests. Try removing it.

Some other thoughts:
1. Idk if the AI reads release notes, but even if it does being a year+ out of date will likely overwhelm its context
2. Being a year out of date makes handling updates as necessary difficult and leaves you exposed to any issues resolved over the previous year as well as increasing the risk and difficulty in updating when it becomes necessary

Reply:
it's guessed `ProvideEnabled` -> `WithServerEnabled`; was there an appropriate path to migrate from `ProvideEnabled` to something else?

Reply:
I made the suggested change in https://github.com/filecoin-project/lotus/pull/13202/commits/69045f4eeaa7440539b752a886798111138a5377 and tests pass.  

I'll give this approval @rvagg so you're able to merge.

Reply:
> Being a year out of date makes handling updates as necessary difficult and leaves you exposed to any issues resolved over the previous year as well as increasing the risk and difficulty in updating when it becomes necessary

Agreed.  There is definitely dependency debt in Lotus.  The process we have talked about is doing dependency updates in bulk after a network upgrade.  That was added to the network upgrade template, but there just hasn't been another network dependency to follow that step.

Reply:
```suggestion
```